core/domain/story_services.py

```diff
@@ -187,33 +189,21 @@ def apply_change_list(story_id, change_list):
         raise
 
 
-def _save_story(committer_id, story, commit_message, change_list):
-    """Validates a story and commits it to persistent storage. If
-    successful, increments the version number of the incoming story domain
-    object by 1.
+def validate_explorations(exp_ids, raise_error):
+    """Validates the explorations in the given story and checks whether they
+    are compatible on the mobile app.
 
     Args:
-        committer_id: str. ID of the given committer.
-        story: Story. The story domain object to be saved.
-        commit_message: str. The commit message.
-        change_list: list(StoryChange). List of changes applied to a story.
+        exp_ids: list(str). The exp IDs to validate.
+        raise_error: bool. Whether to raise an error when an issue is
+            encountered or not. If not, a list of the issues are returned.
+            raise_error should be True when this is called before saving the
+            story and False when this function is called fromn the frontend.
 
-    Raises:
-        ValidationError: An invalid exploration was referenced in the
-            story.
-        Exception: The story model and the incoming story domain
-            object have different version numbers.
+    Returns:
+        list(str). The various validation issues (if raise_error is True).
```

Comment:
Here and elsewhere (in the controllers too), let's call these "validation error messages". When you say issues I think immediately of an Issue domain object. "Messages" are more obviously strings.

Reply:
Done